client/src/helpers/fetch/user/useRegisterMutation.ts

```diff
@@ -0,0 +1,27 @@
+import { useMutation } from "@tanstack/react-query";
+import { NewUser } from "../../../components/Register/types/new-user.types";
+import { User } from "../../../types/user.types";
+
+const baseUrl = "http://localhost:5000"; // TODO: generalize this in an .env variable
+
+// TODO: generalize these fetch functions to prevent duplicating options a
+// million times.
+async function postRegister(newUser: NewUser) {
+	return (
+		await fetch(`${baseUrl}/user/register`, {
+			method: "POST",
+			credentials: "include",
+			headers: {
+				Accept: "application/json",
+				"Content-Type": "application/json",
+			},
+			body: JSON.stringify({ newUser }),
+		})
+	).json();
+}
+
+export default function useRegisterMutation() {
+	return useMutation<User, any, NewUser>(["register"], async (newUser) =>
```

Comment:
Maybe TData should be `User?` instead of `User`, but we do (or should) send an error response from the backend if there was no user created, so maybe not.